File: csharp/Year2015Day01.cs
namespace Aoc;

public static class Year2015Day01
{
    public static void Run()
    {
        var input = ReadInput();
        Run(input);
    }

    public static void Run(string s)
    {
        var floor = 0;
        var firstTimeInBasement = 0;
        var moves = 0;
        var chars = s.ToCharArray();

        foreach (var c in chars)
        {
            moves += 1;
            if (c == '(')
            {
                floor += 1;
            }
            else
            {
                floor -= 1;
                if (firstTimeInBasement == 0 && floor < 0)
                {
                    firstTimeInBasement = moves;
                }
            }
        }

        Console.WriteLine(floor);
        Console.WriteLine(firstTimeInBasement);
    }

    private static string ReadInput()
    {
        return System.IO.File.ReadAllText("../input/2015-01.txt");
    }
}

File: python/year2015day01.py
#!/usr/bin/env python

def run(s):
    floor = 0
    firstTimeInBasement = 0
    moves = 0
    for c in s:
        moves += 1
        if c == '(':
            floor += 1
        else:
            floor -= 1
            if firstTimeInBasement == 0 and floor < 0:
                firstTimeInBasement = moves
    print(floor)
    print(firstTimeInBasement)
